packages/compose/src/hooks/use-debounce/index.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * useDebounce hook
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+
+/**
+ * Use to debounce a value that is changing too frequently.
+ *
+ * Example usage:
+ *
+ *	// An example using the useDebounce hook with useResizeObserver()
+ *  // to debounce a rapidly changing width value
+ *
+ *	const [ resizeListener, sizes ] = useResizeObserver();
+ *	const debouncedSize = useDebounce(sizes.width, 100);
+ *
+ *	useEffect( () => {
+ *
+ *		// ... do your thing here ...
+ *
+ *	}, [debouncedSize] );
+ *
+ * @param {Object}        value  The value changing you want to debounce.
+ * @param {number}        delay  The amount to delay time in ms.
+ * @return {Object}				 Debounced value
```

Comment:
```suggestion
 * @param {*}      value  The value changing you want to debounce.
 * @param {number} delay  The amount to delay time in ms.
 *
 * @return {*} Debounced value.
```

- `value` can be anything, not just extending `Object`
- Mixed whitespace (tabs before parameter name)
- `@return` should be separate grouping, i.e. separated by newline, not aligned with `@param`s ([documentation](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/#functions))

Reply:
If you wanted to get a little fancy, you could use the `@template` tag to reinforce that the type of the return value matches the type of the input value.

```suggestion
 * @template T
 *
 * @param {T}      value  The value changing you want to debounce.
 * @param {number} delay  The amount to delay time in ms.
 *
 * @return {T} Debounced value.
```

See: https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#template